### PR TITLE
[FEAT] Add new rewards formula

### PR DESCRIPTION
**Pull request by kgarbacinski**
New features:
- Add PPF
- Add Comunity Fund
- Change TR & IR
- Change New Matched Strategy
- Change User Budget Strategy
- Align with UTs
- Change default epoch settings

### CODE REVIEW

File: backend/app/infrastructure/database/models.py [hunk 1]
```diff
@@ -110,6 +120,16 @@ class FinalizedEpochSnapshot(BaseModel):
     leftover = Column(db.String, nullable=False)
     withdrawals_merkle_root = Column(db.String)
     total_withdrawals = Column(db.String)
+    ppf = Column(db.String, nullable=True)  # TBC is it needed?
```
Comment (leoni-q) [inline backend/app/infrastructure/database/models.py:123]: I think it's not needed if it's already stored in a pending snapshot. These values won't change through the allocation window

File: backend/app/modules/octant_rewards/core.py [hunk 2]
```diff
@@ -1,17 +1,29 @@
...
 from app.engine.octant_rewards import OctantRewardsSettings
+from app.engine.octant_rewards.community_fund import CommunityFundPayload
 from app.engine.octant_rewards.locked_ratio import LockedRatioPayload
 from app.engine.octant_rewards.operational_cost import OperationalCostPayload
+from app.engine.octant_rewards.ppf import PPFPayload
 from app.engine.octant_rewards.total_and_individual import TotalAndAllIndividualPayload
 
 
+@dataclass
+class OctantRewardsWrapper:
```
Comment (leoni-q) [inline backend/app/modules/octant_rewards/core.py:13]: 👍

File: backend/tests/engine/test_epoch_settings.py [hunk 3]
```diff
@@ -18,70 +22,103 @@
...
     settings = get_epoch_settings(-1)
     check_settings(
-        settings,
+        settings=settings,
         total_and_all_individual_rewards=DefaultTotalAndAllIndividualRewards(),
         timebased_weights=TimebasedWithoutUnlocksWeights(),
         operational_cost=OpCostPercent(Decimal("0.25")),
+        ppf=PPFCalculatorPercent(OctantRewardsDefaultValues.PPF),
+        community_fund=CommunityFundPercent(OctantRewardsDefaultValues.COMMUNITY_FUND),
+        user_budget=DefaultUserBudget(),
```
Comment (leoni-q) [inline backend/tests/engine/test_epoch_settings.py:47]: default settings for upcoming epochs should be the same as settings for epoch 3.

File: backend/tests/helpers/constants.py [hunk 4]
```diff
@@ -3,7 +3,9 @@
 MNEMONIC = "test test test test test test test test test test test junk"
 MOCKED_PENDING_EPOCH_NO = 1
 MOCKED_FINALIZED_EPOCH_NO = 1
+MOCKED_EPOCH_NO_AFTER_OVERHAUL = 3
 MOCKED_CURRENT_EPOCH_NO = 2
+NO_PATRONS_REWARDS = 0
```
Comment (leoni-q) [inline backend/tests/helpers/constants.py:8]: I'm not sure about this - what's the point in adding this const if it's set to zero?

Comment (kgarbacinski) [reply backend/tests/helpers/constants.py:8]: NO_PATRONS_REWARDS is a const representing that there's lack of patrons_rewards. I've extracted it to use it in different formulas explicitly to roughly show that a given formula uses PATRONS_REWARDS value but in the cases it's exceptionally set to 0. 

Example:
MATCHED_REWARDS = TOTAL_REWARDS - ALL_INDIVIDUAL_REWARDS --- without NO_PATRONS_REWARDS
MATCHED_REWARDS = TOTAL_REWARDS - ALL_INDIVIDUAL_REWARDS + NO_PATRONS_REWARDS ---- now, with NO_PATRONS_REWARDS

WDYT?

File: backend/app/infrastructure/database/models.py (hunk 1 above)
Comment (kgarbacinski) [reply backend/app/infrastructure/database/models.py:123]: Okay, will get a rid of it.

File: backend/app/engine/octant_rewards/__init__.py [hunk 5]
```diff
@@ -1,25 +1,47 @@
...
     )
     matched_rewards: MatchedRewards = field(default_factory=DefaultMatchedRewards)
+
+    ppf: PPFCalculator = field(
+        default_factory=lambda: PPFCalculatorPercent(OctantRewardsDefaultValues.PPF)
+    )
+    community_fund: CommunityFundCalculator = field(
+        default_factory=lambda: CommunityFundPercent(
+            OctantRewardsDefaultValues.COMMUNITY_FUND
+        )
```
Comment (0xartem) [inline backend/app/engine/octant_rewards/__init__.py:46]: Do we have a field for increased staking here? @leoni-q @kgarbacinski  I think we used it in the epoch 2 and might also use it in the future, we can set it to 0 for epoch 3 and going forward for now.

**Review by 0xartem — APPROVED** [review]

File: backend/app/engine/octant_rewards/__init__.py (hunk 5 above)
Comment (leoni-q) [reply backend/app/engine/octant_rewards/__init__.py:46]: we have only `leftover` currently, which includes `increased_staking` and the rest what's left. this story might be good to introduce `increased_staking` concept and decide how we calculate it: https://linear.app/golemfoundation/issue/OCT-1364/metrics-funds-usage-for-e2-should-show-352-eth-for-staking

**Comment by leoni-q** [comment]
In order to apply a new migration please run `flask db migrate -m "..."` and commit a migration file  to this PR

**Review by leoni-q — APPROVED** [review]

**Pull request merged**